src/Internal/Payment/PaymentContext.php

```diff
@@ -7,9 +7,92 @@
 
 namespace WCPay\Internal\Payment;
 
+use Automattic\WooCommerce\Blocks\Payments\PaymentMethodTypeInterface;
+use WCPay\Internal\Payment\PaymentMethod\PaymentMethodInterface;
+
 /**
  * A context object, which is shared between payment states.
  */
 class PaymentContext {
+	/**
+	 * Contains all the context's data.
+	 *
+	 * @var array
+	 */
+	private $data = [];
+
+	/**
+	 * Stores an internal value.
+	 * Use this method for changes to allow logging in the future.
+	 *
+	 * @param string $key   Property name.
+	 * @param mixed  $value Value to store.
+	 */
+	private function set( string $key, $value ) {
```

Comment:
Nitpick. Do we plan to use PSR12 for function/variable naming as we use PSR4 for class naming (for the sake of autoloader)?